staging/src/k8s.io/component-helpers/resource/helpers.go

```diff
@@ -44,15 +45,70 @@ type PodResourcesOptions struct {
 	// NonMissingContainerRequests if provided will replace any missing container level requests for the specified resources
 	// with the given values.  If the requests for those resources are explicitly set, even if zero, they will not be modified.
 	NonMissingContainerRequests v1.ResourceList
+	// PodLevelResourcesEnabled indicates that the pod-level resources feature gate
+	// is enabled.
+	PodLevelResourcesEnabled bool
+	// UseContainerLevelResources controls whether pod resource calculation should use
+	// container-level resources to calculate PodRequests even when
+	// PodLevelResources feature is enabled and pod-level resources are set.
+	UseContainerLevelResources bool
```

Comment:
I think setting this option `UseContainerLevelResources` caused the confusion from @haircommander in https://github.com/kubernetes/kubernetes/pull/128407/commits/56235c8d4f38bc2c1884e54b1c50122870fd3ab9#r1825874699. 

The source of the confusion probably comes from the fact that we reuse the same function for different purposes and simply set these boolean values to distinguish them. 

Some thoughts since this function is considered an API: 
 - Most components should not need to know when to set this option. It's not set by a feature gate and is not clear. 
 - Create another function for the use case where `UseContainerLevelResources` is always true. Components that know to call this will call this new function directly.

This likely is an overkill given that the list of fields in `PodResourcesOptions` is already very long with in-depth knowledge required to set them. It seems more like a utility function than an API in any way.

Tagging the API reviewers to see if they have any thought @thockin @cici37 
Otherwise, feel free to resolve this comment

Reply:
I prefer adding a separate function AggregateContainerRequests. The only problem is that it needs some params from PodResourcesOptions. Right now AggregateContainerRequests accepts PodResourceOptions. Any suggestion for this?

Also, I wanted to move AggregateContainerRequests to an internal helper function, but I don't see any internal helpers being imported in staging/src/k8s.io. Is there a standard around this?